File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 2

  def self.compute(strand_one, strand_two)
    verify_equivalent_lengths(strand_one, strand_two)

    (0..strand_one.length).count do |i|
      strand_one[i] != strand_two[i]
    end
  end

  def self.verify_equivalent_lengths(strand_one, strand_two)
    raise ArgumentError unless strand_one.length == strand_two.length
  end
end

File: javascript/hamming/hamming.js
var Hamming = function(){
};

var Error = function(message){
  this.message = message;
};

function checkLengthsEqualBetween(strandOne, strandTwo){
  return (strandOne.length === strandTwo.length);
};

function calcHammingDistance(strandOne, strandTwo){
  var hammingDistance = 0;
  for (i = 0; i < strandOne.length; i++) {
    if (strandOne[i] !== strandTwo[i]) {
      hammingDistance += 1;
    }
  };
  return hammingDistance;
};

Hamming.prototype.compute = function (strandOne, strandTwo){
  if (checkLengthsEqualBetween(strandOne, strandTwo)){
    return calcHammingDistance(strandOne, strandTwo);
  } else {
    throw new Error("DNA strands must be of equal length.");
  }
};

module.exports = Hamming;
